enr/enr.go

```diff
@@ -27,7 +27,8 @@ type Record struct {
 	Sequence  uint64 // The sequence number. Nodes should increase the number whenever the record changes and republish the record.
 	IDScheme  string // name of identity scheme, e.g. “v4”
 
-	PublicKey *secp256k1.PublicKey
+	PrivateKey *secp256k1.PrivateKey
```

Comment:
I can see why you'd like the PrivateKey here for ease-of-use, but is it a good idea for the ENR struct abstraction? This would only be non-nil to the local record right?

Reply:
Good call. I was thinking more about this package. I think I'd like to rename it to `node` and refactor it a bit. As I start building the cmd, I've been wanting a node struct that holds on to: prv(for local)/pub keys, tcp/udp info, and the actually connection. And of course you can marshal / un-marshal enr records from the node.

Reply:
This isn't ideal, but I'll fix this in a subsequent commit where we beef up the enr package and rename it to node